src/main/scala/org/broadinstitute/hail/driver/GroupTestSKATO.scala

```diff
@@ -0,0 +1,359 @@
+
+    @Args4jOption(required = false, name = "--method",
+      usage = "Method for calculating p-values. One of [davies, liu, liu.mod, optimal.adj, optimal]")
+    var method: String = "davies"
+
+    @Args4jOption(required = false, name = "--weights-beta",
+      usage = "Comma-separated parameters for beta function for calculating weights. Default is 1,25")
+    var weightsBeta: String = "1,25"
+
+    @Args4jOption(required = false, name = "--impute-method",
+      usage = "Method for imputing missing genotypes. One of [fixed, random, bestguess]")
+    var imputeMethod: String = "fixed"
+
+    @Args4jOption(required = false, name = "--r-corr",
+      usage = "The rho parameter for the unified test. rho=0 is SKAT only. rho=1 is Burden only. Can also be multiple comma-separated values")
+    var rCorr: String = "0.0"
+
+    @Args4jOption(required = false, name = "--missing-cutoff",
+      usage = "Missing rate cutoff for variant inclusion")
+    var missingCutoff: Double = 0.15
+
+    @Args4jOption(required = false, name = "--estimate-maf",
+      usage = "Method for estimating MAF. 1 = use all samples to estimate MAF. 2 = use samples with non-missing phenotypes and covariates")
+    var estimateMAF: Int = 1
+
+  }
+
+  def newOptions = new Options
+
+  def name = "grouptest skato"
+
+  def description = "Run SKAT-O on groups specified by a variant annotation"
+
+  def supportsMultiallelic = false
+
+  def requiresVDS = true
+
+  val skatoElementSignature = TStruct(
+    "groupName" -> TInt,
+    "pValue" -> TDouble,
+    "pValueNoAdj" -> TDouble,
+    "nMarker" -> TInt,
+    "nMarkerTest" -> TInt
+  )
+
+  val skatoSignature = TArray(skatoElementSignature)
+
+  val header = "groupName\tpValue\tpValueNoAdj\tnMarker\tnMarkerTest"
+
+  def run(state: State, options: Options): State = {
+    val vds = state.vds
+    val sampleIds = vds.sampleIds
+
+    val properties = try {
+      val p = new Properties()
+      val is = new FileInputStream(options.config)
+      p.load(is)
+      is.close()
+      p
+    } catch {
+      case e: IOException =>
+        fatal(s"could not open file: ${e.getMessage}")
+    }
+
+    val yType = if (options.quantitative) "C" else "D"
+    val nResampling = options.nResampling
+    val typeResampling = options.typeResampling
+    val adjustment = !options.noAdjustment
+    val kernel = options.kernel
+    val method = options.method
+    val weightsBeta = options.weightsBeta
+    val imputeMethod = options.imputeMethod
+    val rCorr = options.rCorr
+    val missingCutoff = options.missingCutoff
+    val estimateMAF = options.estimateMAF
+    val quantitative = options.quantitative
+    val splat = options.splat
+    val groupNameSep = options.groupNameSep
+
+    val R = properties.getProperty("hail.skato.Rscript", "/usr/bin/Rscript")
+    val skatoScript = properties.getProperty("hail.skato.script", "src/dist/scripts/skato.r")
+
+    val rCorrArray = options.rCorr.split(",").map(_.toDouble)
```

Comment:
`.toDouble` can throw `java.lang.NumberFormatException`.  You need to handle it.  Same in a number of places below.